### PR TITLE
chore(tracing): deprecate DD_TRACE_ANALYTICS_ENABLED

### PR DESCRIPTION
**Pull request by mabdinur**
Datadog App Analytics was deprecated years ago. We can deprecate this configuration and remove this feature in v3.0: https://docs.datadoghq.com/tracing/legacy_app_analytics/

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/settings/integration.py [hunk 1]
```diff
@@ -67,11 +69,18 @@ def _get_analytics_settings(self):
         # Set default analytics configuration, default is disabled
         # DEV: Default to `None` which means do not set this key
         # Inject environment variables for integration
-        _ = os.getenv(
-            "DD_TRACE_%s_ANALYTICS_ENABLED" % self.integration_name.upper(),
-            os.getenv("DD_%s_ANALYTICS_ENABLED" % self.integration_name.upper()),
-        )
-        analytics_enabled = asbool(_) if _ is not None else None
+        env = "DD_TRACE_%s_ANALYTICS_ENABLED" % self.integration_name.upper(),
```
Comment (mabdinur) [inline ddtrace/settings/integration.py:72]:
```suggestion
        env = "DD_TRACE_%s_ANALYTICS_ENABLED" % self.integration_name.upper()
```

File: ddtrace/constants.py [hunk 2]
```diff
@@ -47,3 +51,21 @@
...
 # Use this to explicitly inform the backend that a trace should be kept and stored.
 USER_KEEP = 2
+
+
+_DEPRECATED_MODULE_ATTRIBUTES = [
+    "ANALYTICS_SAMPLE_RATE_KEY",
+]
+
+
+def __getattr__(name):
```
Comment (emmettbutler) [inline ddtrace/constants.py:61]: This is a really nice way to handle the deprecation

**Review by emmettbutler — APPROVED** [review]

File: ddtrace/constants.py [hunk 3]
```diff
@@ -47,3 +51,21 @@
...
+    if name in _DEPRECATED_MODULE_ATTRIBUTES:
+        _debtcollector.deprecate(
+            ("%s.%s is deprecated" % (__name__, name)),
+            category=_DDTraceDeprecationWarning,
```
Comment (mabdinur) [inline ddtrace/constants.py:65]:
```suggestion
            category=_DDTraceDeprecationWarning,
            version="3.0.0"
```

File: releasenotes/notes/deprecate-app-analytics-9718797d7a788275.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+deprecations:
+  - |
+    tracing: Deprecates support for `APM Legacy App Analytics <https://docs.datadoghq.com/tracing/legacy_app_analytics/>`_.
+    This feature and of its associated configuration options are deprecated and will be removed in a future release.
```
Comment (ZStriker19) [inline releasenotes/notes/deprecate-app-analytics-9718797d7a788275.yaml:5]:
```suggestion
    This feature and its associated configuration options are deprecated and will be removed in a future release.
```

**Review by ZStriker19 — APPROVED** [review]

**Review by wconti27 — APPROVED** [review]

**Pull request merged**